components/eam/src/physics/cam/physpkg.F90

```diff
@@ -1776,7 +1779,7 @@ subroutine tphysac (ztodt,   cam_in,  &
 
     ! DCAPE-ULL: record current state of T and q for computing dynamical tendencies
     !            the calculation follows the same format as in diag_phys_tend_writeout
-    if (do_zmconv_dcape_ull .or. do_zmconv_dcape_only) then
+    if (deep_scheme .eq. 'ZM' .and. (do_zmconv_dcape_ull .or. do_zmconv_dcape_only)) then
```

Comment:
Since this is a string comparison, do we need `trim(adjustl(deep_scheme)) .eq. 'ZM'`?

Reply:
Good point. But I think it is ok. It will force users to specify just 'ZM' as the scheme name. (it is currently set so as default). Existing codes, say in convect_deep.F90, also use it as is without trimming. Thanks for the quick review, @singhbalwinder .